event/oracle/upgrade_oracle_event.go

```diff
@@ -0,0 +1,178 @@
+	return "'OracleUpgrade'"
+}
+
+func (e UpgradeOracleEvent) EventHandler(event ctypes.ResultEvent) error {
+	uniqueID := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyUniqueID][0]
+	addressValue := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyOracleAddress][0]
+	queryClient := e.reactor.QueryClient()
+
+	oracleRegistration, err := queryClient.GetOracleRegistration(addressValue, uniqueID)
+	if err != nil {
+		log.Infof("failed to get oracleRegistration, voting ignored. uniqueID(%s), address(%s). %v", uniqueID, addressValue, err)
+		return err
+	}
+
+	voteOption, err := e.verifyAndGetVoteOption(oracleRegistration)
+	if err != nil {
+		switch voteOption {
+		case oracletypes.VOTE_OPTION_NO:
+			log.Infof("vote No due to error while verify: %v", err)
+		case oracletypes.VOTE_OPTION_UNSPECIFIED:
+			log.Errorf("can't vote due to error whiile verify. uniqueID(%s), votingTargetAddress(%s), error(%v)",
+				uniqueID,
+				addressValue,
+				err)
+			return err
+		default:
+			log.Warnf("if an error occurs, no other voteOption is possible. uniqueID(%s), votingTargetAddress(%s) voteOption(%s), err(%v)",
+				uniqueID,
+				addressValue,
+				voteOption,
+				err)
+		}
+	}
+
+	msgVoteOracleRegistration, err := makeOracleRegistrationVote(
+		uniqueID,
+		e.reactor.OracleAcc().GetAddress(),
+		addressValue,
+		voteOption,
+		e.reactor.OraclePrivKey().Serialize(),
+		oracleRegistration.NodePubKey,
+		oracleRegistration.Nonce,
+	)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("oracle upgrade voting info. uniqueID(%s), votingTargetAddress(%s), voteOption(%s)",
+		msgVoteOracleRegistration.OracleRegistrationVote.UniqueId,
+		msgVoteOracleRegistration.OracleRegistrationVote.VotingTargetAddress,
+		msgVoteOracleRegistration.OracleRegistrationVote.VoteOption,
+	)
+
+	txBuilder := panacea.NewTxBuilder(*e.reactor.QueryClient())
+
+	txBytes, err := generateTxBytes(msgVoteOracleRegistration, e.reactor.OracleAcc().GetPrivKey(), e.reactor.Config(), txBuilder)
+	if err != nil {
+		return err
+	}
+
+	if err := broadcastTx(e.reactor.GRPCClient(), txBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e UpgradeOracleEvent) verifyAndGetVoteOption(r *oracletypes.OracleRegistration) (oracletypes.VoteOption, error) {
+	upgradeInfo, err := e.reactor.QueryClient().GetOracleUpgradeInfo()
+	if err != nil {
+		if errors.Is(err, panacea.ErrEmptyValue) {
+			return oracletypes.VOTE_OPTION_NO, fmt.Errorf("not found oracle upgrade info. %w", err)
+		}
+		return oracletypes.VOTE_OPTION_UNSPECIFIED, fmt.Errorf("failed to get oracle upgrade info. %v", err)
+	}
+	if upgradeInfo.UniqueId != r.UniqueId {
+		return oracletypes.VOTE_OPTION_NO, fmt.Errorf("oracle's uniqueID does not match the uniqueID being upgraded. expected uniqueID(%s), oracle's uniqueID(%s), ",
+			upgradeInfo.UniqueId,
+			r.UniqueId)
+	}
+
+	block, err := e.reactor.QueryClient().GetLightBlock(r.TrustedBlockHeight)
```

Comment:
Is there a reason for verifying the light block here? Is it already verified in queryClient?

Reply:
Your question is why validate block information in here?

Reply:
If that's light,
When oracle's will be upgrade, it is changing uniqueID.
So the old lightClient data cannot used.

We used a new lightClient data in new oracle binary.

Reply:
Ah.. sorry. There is light block info in `OracleRegistration`. So, we have to validate it... I had misunderstood.

Reply:
Aha. There is another reason. Now I understand. thank you